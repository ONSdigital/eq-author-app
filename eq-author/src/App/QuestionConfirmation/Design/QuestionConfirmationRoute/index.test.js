import React from "react";
import { shallow } from "enzyme";

import { UnwrappedQuestionConfirmationRoute as QuestionConfirmationRoute } from "./";

describe("QuestionConfirmationRoute", () => {
  let mockHandlers;

  const defaultProps = {
    loading: false,
    data: {
      questionConfirmation: {
        id: "1",
        displayName: "My first displayname",
        title: "My first confirmation",
        page: {
          id: "1",
          displayName: "My question",
          answers: []
        },
        positive: {
          label: "Positive label",
          description: "Positive description"
        },
        negative: {
          label: "Negative label",
          description: "Negative description"
        }
      }
    }
  };

  const render = props => {
    return shallow(
      <QuestionConfirmationRoute
        {...mockHandlers}
        {...defaultProps}
        {...props}
      />
    );
  };

  beforeEach(() => {
    mockHandlers = {
      onUpdateQuestionConfirmation: jest.fn(),
      onDeleteQuestionConfirmation: jest.fn()
    };
  });

  it("should render", () => {
    expect(render()).toMatchSnapshot();
  });

  it("should show loading info when loading", () => {
    expect(render({ loading: true })).toMatchSnapshot();
  });

  it("should show error info when there is an error", () => {
    expect(render({ error: { message: "some error" } })).toMatchSnapshot();
  });

  it("should update the confirmation page when requested to", () => {
    const wrapper = render();
    wrapper
      .find(`[data-test="editor"]`)
      .simulate("update", "Updated question confirmation");
    expect(mockHandlers.onUpdateQuestionConfirmation).toHaveBeenCalledWith(
      "Updated question confirmation"
    );
  });

  it("should delete the confirmation when the toolbar delete button is clicked", () => {
    const wrapper = render();
    wrapper.find(`[data-test="btn-delete"]`).simulate("click");
    expect(mockHandlers.onDeleteQuestionConfirmation).toHaveBeenCalledWith(
      defaultProps.data.questionConfirmation
    );
  });
});
