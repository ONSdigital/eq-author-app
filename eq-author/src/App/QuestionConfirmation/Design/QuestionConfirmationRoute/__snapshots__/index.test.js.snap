// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionConfirmationRoute should render 1`] = `
<EditorLayout>
  <EditorToolbar__Toolbar>
    <EditorToolbar__Buttons>
      <IconButtonDelete
        data-test="btn-delete"
        hideText={false}
        onClick={[Function]}
      >
        Delete
      </IconButtonDelete>
    </EditorToolbar__Buttons>
  </EditorToolbar__Toolbar>
  <Connect(withEntityEditor(UnwrappedEditor))
    confirmation={
      Object {
        "displayName": "My first displayname",
        "id": "1",
        "negative": Object {
          "description": "Negative description",
          "label": "Negative label",
        },
        "page": Object {
          "answers": Array [],
          "displayName": "My question",
          "id": "1",
        },
        "positive": Object {
          "description": "Positive description",
          "label": "Positive label",
        },
        "title": "My first confirmation",
      }
    }
    data-test="editor"
    onUpdate={[MockFunction]}
  />
</EditorLayout>
`;

exports[`QuestionConfirmationRoute should show error info when there is an error 1`] = `
<EditorLayout>
  <Error>
    Oh no! Something went wrong
  </Error>
</EditorLayout>
`;

exports[`QuestionConfirmationRoute should show loading info when loading 1`] = `
<EditorLayout>
  <Loading
    height="100%"
  >
    Confirmation is loading
  </Loading>
</EditorLayout>
`;
